app/assets/javascripts/discourse/app/components/d-editor.js

```diff
@@ -413,92 +407,82 @@ export default Component.extend(TextareaTextManipulation, {
     return toolbar;
   },
 
-  cachedCookAsync(text) {
-    if (this._cachedCookFunction) {
-      return Promise.resolve(this._cachedCookFunction(text));
-    }
+  async cachedCookAsync(text, options = {}) {
+    this._cachedCookFunction ||= await generateCookFunction(options || {});
+    return await this._cachedCookFunction(text);
+  },
 
-    const markdownOptions = this.markdownOptions || {};
-    return generateCookFunction(markdownOptions).then((cook) => {
-      this._cachedCookFunction = cook;
-      return cook(text);
-    });
+  async morph(dst, src) {
+    this._morphFunction ||= (await import("morphlex")).morph;
+    return await this._morphFunction(dst, src);
   },
 
-  _updatePreview() {
-    if (this._state !== "inDOM" || !this.processPreview) {
+  async _updatePreview() {
+    if (
+      this._state !== "inDOM" ||
+      !this.processPreview ||
+      this.isDestroying ||
+      this.isDestroyed
+    ) {
       return;
     }
 
-    const value = this.value;
+    const cooked = await this.cachedCookAsync(this.value, this.markdownOptions);
 
-    this.cachedCookAsync(value).then((cooked) => {
-      if (this.isDestroyed) {
-        return;
-      }
+    if (this.preview === cooked) {
+      return;
+    }
 
-      if (this.preview === cooked) {
-        return;
-      }
+    this.set("preview", cooked);
 
-      this.set("preview", cooked);
+    if (this.siteSettings.enable_diffhtml_preview) {
+      const cookedElement = document.createElement("div");
+      cookedElement.className = this._previewElement.className;
+      cookedElement.innerHTML = cooked;
 
-      let previewPromise = Promise.resolve();
+      // Same order of operation as in the "previewUpdated" method in "composer-editor.js"
```

Comment:
Kept the same order of operation than in the `previewUpdated` method in the `composer-editor.js` file to be more consistent.